diff --git a/go.mod b/go.mod
index f6914a8..95b3845 100644
--- a/go.mod
+++ b/go.mod
@@ -8,7 +8,7 @@ require (
        github.com/pkg/errors v0.9.1
        github.com/tjfoc/gmsm v1.4.1 // indirect
        github.com/urfave/cli v1.21.0
-       github.com/xtaci/kcp-go/v5 v5.6.1
+       github.com/xtaci/kcp-go/v51 v51.0.0 //v5.6.1
        github.com/xtaci/smux v1.5.16
        github.com/xtaci/tcpraw v1.2.25
        golang.org/x/crypto v0.0.0-20220622213112-05595931fe9d
diff --git a/vendor/github.com/xtaci/kcp-go/v5/fec.go b/vendor/github.com/xtaci/kcp-go/v5/fec.go
index 0a203ee..aec48b7 100644
--- a/vendor/github.com/xtaci/kcp-go/v5/fec.go
+++ b/vendor/github.com/xtaci/kcp-go/v5/fec.go
@@ -10,8 +10,8 @@ import (
 const (
        fecHeaderSize      = 6
        fecHeaderSizePlus2 = fecHeaderSize + 2 // plus 2B data size
-       typeData           = 0xf1
-       typeParity         = 0xf2
+       typeData           = 0xa001
+       typeParity         = 0xa002
        fecExpire          = 60000
        rxFECMulti         = 3 // FEC keeps rxFECMulti* (dataShard+parityShard) ordered packets in memory
 )
diff --git a/vendor/github.com/xtaci/kcp-go/v5/go.mod b/vendor/github.com/xtaci/kcp-go/v5/go.mod
index ff51020..4e8d50f 100644
--- a/vendor/github.com/xtaci/kcp-go/v5/go.mod
+++ b/vendor/github.com/xtaci/kcp-go/v5/go.mod
@@ -1,4 +1,4 @@
-module github.com/xtaci/kcp-go/v5
+module github.com/xtaci/kcp-go/v51

 require (
        github.com/klauspost/cpuid v1.3.1 // indirect
diff --git a/vendor/modules.txt b/vendor/modules.txt
index b5be4c4..ae949f1 100644
--- a/vendor/modules.txt
+++ b/vendor/modules.txt
@@ -26,9 +26,9 @@ github.com/tjfoc/gmsm/sm4
 # github.com/urfave/cli v1.21.0
 ## explicit
 github.com/urfave/cli
-# github.com/xtaci/kcp-go/v5 v5.6.1
+# github.com/xtaci/kcp-go/v51 v51.0.0
 ## explicit
-github.com/xtaci/kcp-go/v5
+github.com/xtaci/kcp-go/v51
 # github.com/xtaci/smux v1.5.16
 ## explicit
 github.com/xtaci/smux
root@VM-8-8-ubuntu:~/newkcp/kcptun# git commit -m "my"
HEAD detached at v20221015
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   go.mod
        modified:   vendor/github.com/xtaci/kcp-go/v5/fec.go
        modified:   vendor/github.com/xtaci/kcp-go/v5/go.mod
        modified:   vendor/modules.txt

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        mybuild.sh

no changes added to commit (use "git add" and/or "git commit -a")
